packages/devtools_app/lib/src/routing.dart

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+const appSizePageId = 'app-size';
+const homePageId = '';
+const snapshotPageId = 'snapshot';
+
+/// Represents a Page/route for a DevTools screen.
+class DevToolsRouteConfiguration {
+  DevToolsRouteConfiguration(this.page, this.args);
+
+  final String page;
+  final Map<String, String> args;
+}
+
+/// Converts between structured DevToolsRouteConfiguration (our internal data
+/// for pages/routing) and RouteInformation (generic data that can be persisted
+/// in the address bar/state objects).
+class DevToolsRouteInformationParser
+    extends RouteInformationParser<DevToolsRouteConfiguration> {
+  @override
+  Future<DevToolsRouteConfiguration> parseRouteInformation(
+      RouteInformation routeInformation) {
+    final uri = Uri.parse(routeInformation.location);
+    final path = uri.path.isNotEmpty ? uri.path.substring(1) : '';
+    final configuration = DevToolsRouteConfiguration(path, uri.queryParameters);
+    return SynchronousFuture<DevToolsRouteConfiguration>(configuration);
```

Comment:
why SynchronousFuture and not just Future?

Reply:
It avoids waiting for the next microtask (the docs for the base class recommend doing this where possible https://github.com/flutter/flutter/blob/d65e2c3a27ca566b2e276725588315830106fc0a/packages/flutter/lib/src/widgets/router.dart#L112-L116).